### PR TITLE
Updates from feedbacks

### PR DESCRIPTION
**Pull request by zdenal**
- [x] make section about audience expression in README

- [x] make test returning `0`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,28 +90,9 @@ So if we say from our sample 2 women and 1 man (with same weightings) are using
...
-  percentage: 41.18
-},
-{
-  option: 'female',
-  responses_count: 2,
-  weighted: 200,
-  percentage: 58.82
-}
-```
-Now we can say that 41.18% of iPhone users are male and 58.82% are female.
```
Comment (zdenal) [inline README.md]: Do we need this example again? I think in ` #### 1.1. Why do we have weightings?` it is described enough. Trying to make README not so big with needed information and not to look big and difficult.

**Comment by romansklenar** [comment]
So far looks good 👌

**Comment by zdenal** [comment]
Hi @romansklenar I added test with zero results and made `Audience expression` section. Could you please check if it is ok? Thanks

**Comment by zdenal** [comment]
@romansklenar also this PR at all is RDY.

File: README.md [hunk 2]
```diff
@@ -18,9 +18,24 @@ The analysis include calculated metrics.
 * **responses_count** - the count of respondents related to the option in the audience
 * **weighted** - the sum of respondents weighting related to the option in the audience.
 
+### Audience expression
+Audience expression is used as filter for respondent sample. By this expression we are describing the sample over which we are making calculations. As we can see
+in test the expression has JSON format. So for example expression like this: `{and: [{gender: [:male]}, {age: [:young]}]}` are saying we are interested in **Young men** sample/audince.
+
+Also the expression could be nested (doesn't need to be only one level `and` or `or` array in JSON). The expression could looks like:
```
Comment (romansklenar) [inline README.md]: Typo: 
> The expression could look like this:

File: README.md [hunk 3]
```diff
@@ -18,9 +18,24 @@ The analysis include calculated metrics.
...
+`{and: [{question1: [option1]}, {or: [{question2: [option2]}, {question3: [option3, option4]}]}]}`. The grammar below is describing the structure.
+
+```
+E -> AND | OR | q
+AND -> { and: Array<E> }
+OR -> { or: Array<E> }
+q -> { question_code: Array<question_option> }
```
Comment (romansklenar) [inline README.md]: Suggestion: `Q -> { question: Array<option> }`, then on line 29: `E -> AND | OR | Q`

File: README.md [hunk 4]
```diff
@@ -18,9 +18,24 @@ The analysis include calculated metrics.
 * **responses_count** - the count of respondents related to the option in the audience
 * **weighted** - the sum of respondents weighting related to the option in the audience.
 
+### Audience expression
+Audience expression is used as filter for respondent sample. By this expression we are describing the sample over which we are making calculations. As we can see
+in test the expression has JSON format. So for example expression like this: `{and: [{gender: [:male]}, {age: [:young]}]}` are saying we are interested in **Young men** sample/audince.
+
+Also the expression could be nested (doesn't need to be only one level `and` or `or` array in JSON). The expression could look like this:
+`{and: [{question1: [option1]}, {or: [{question2: [option2]}, {question3: [option3, option4]}]}]}`. The grammar below is describing the structure.
```
Comment (romansklenar) [inline README.md]: 👍

File: spec/calc_spec.rb [hunk 5]
```diff
@@ -85,7 +87,28 @@
...
+            option: 'male',
+            responses_count: 0,
+            weighted: 0,
+            percentage: 0
+          },
+          {
+            option: 'female',
+            responses_count: 0,
+            weighted: 0,
+            percentage: 0
```
Comment (romansklenar) [inline spec/calc_spec.rb]: 👍

File: README.md (hunk 3 above)
Comment (zdenal) [reply README.md]: @romansklenar is there some reason for keep it all same? I wanted distinguish terminals from nonterminals in grammar ....

File: README.md (hunk 2 above)
Comment (zdenal) [reply README.md]: @romansklenar done

File: README.md (hunk 3 above)
Comment (romansklenar) [reply README.md]: OK

**Pull request merged**